pipeline/matcher/src/main/scala/uk/ac/wellcome/platform/matcher/storage/elastic/ElasticWorkLinksRetriever.scala

```diff
@@ -1,29 +1,55 @@
 package uk.ac.wellcome.platform.matcher.storage.elastic
 
+import com.sksamuel.elastic4s.ElasticDsl.get
+import com.sksamuel.elastic4s.requests.get.{GetRequest, GetResponse}
+import com.sksamuel.elastic4s.circe._
 import com.sksamuel.elastic4s.{ElasticClient, Index}
 import uk.ac.wellcome.json.JsonUtil._
-import uk.ac.wellcome.models.work.internal.{Work, WorkState}
-import uk.ac.wellcome.pipeline_storage.{
-  ElasticRetriever,
-  Retriever,
-  RetrieverMultiResult
-}
+import uk.ac.wellcome.models.work.internal.{IdState, MergeCandidate}
+import uk.ac.wellcome.pipeline_storage.elastic.ElasticRetriever
 import uk.ac.wellcome.platform.matcher.models.WorkLinks
 
-import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.ExecutionContext
+import scala.util.Try
 
-class ElasticWorkLinksRetriever(client: ElasticClient, index: Index)(
+/** The matcher only needs to know about linking data on works; most of the fields
+  * it can completely ignore.
+  *
+  * This custom retriever fetches only the specific fields that the matcher needs, to
+  * reduce the amount of data we have to get out of Elasticsearch (and pay to send through
+  * the AWS NAT Gateway).
+  */
```

Comment:
I like it  either way as it's efficient, but would the [`PrivateLink`](https://github.com/wellcomecollection/catalogue/pull/1261) render the NAT gateway reason redundant?

Reply:
Yup.